Sources/BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -39,6 +39,24 @@ public class NetworkStore: ObservableObject {
   private let rpcService: BraveWalletJsonRpcService
   private let walletService: BraveWalletBraveWalletService
   private let swapService: BraveWalletSwapService
+  
+  private weak var networkSelectionStore: NetworkSelectionStore?
+  
+  func openNetworkSelectionStore(mode: NetworkSelectionStore.Mode = .select) -> NetworkSelectionStore {
+    if let store = networkSelectionStore {
+      if store.mode == mode {
+        return store
+      } else {
+        networkSelectionStore = nil
+      }
+    }
+    let store = NetworkSelectionStore(mode: mode, networkStore: self)
+    networkSelectionStore = store
+    return store
+  }
+  func closeNetworkSelectionStore() {
```

Comment:
nit: can these `openNetworkSelectionStore(mode:)` and `closeNetworkSelectionStore()` functions go below the initializer for `NetworkStore`? Fine to move `NetworkSelectionStore` property with them to keep them together 🙂

Reply:
fixed in [16ed6c6](https://github.com/brave/brave-ios/pull/6360/commits/16ed6c61274a19652e7b68b84877edd072fc989a)